osu.Game.Tests/Editing/Checks/CheckAudioQualityTest.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using Moq;
+using NUnit.Framework;
+using osu.Framework.Audio.Track;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit.Checks;
+using osu.Game.Rulesets.Objects;
+
+namespace osu.Game.Tests.Editing.Checks
+{
+    [TestFixture]
+    public class CheckAudioQualityTest
+    {
+        private CheckAudioQuality check;
+        private IBeatmap beatmap;
+
+        [SetUp]
+        public void Setup()
+        {
+            check = new CheckAudioQuality();
+            beatmap = new Beatmap<HitObject>
+            {
+                BeatmapInfo = new BeatmapInfo
+                {
+                    Metadata = new BeatmapMetadata { AudioFile = "abc123.jpg" }
+                }
+            };
+        }
+
+        [Test]
+        public void TestMissing()
+        {
+            // While this is a problem, it is out of scope for this check and is caught by a different one.
+            beatmap.Metadata.AudioFile = null;
+
+            var mock = new Mock<IWorkingBeatmap>();
+            mock.SetupGet(_ => _.Beatmap).Returns(beatmap);
```

Comment:
In this project, `_` imply discards, not usages. Please name these (like `b` or `w` for workingbeatmap, `t` for track, etc).
This is done in multiple places in this PR, so please change all.